mongodbatlas/fw_resource_mongodbatlas_search_node.go

```diff
@@ -0,0 +1,362 @@
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+	"github.com/mongodb/terraform-provider-mongodbatlas/mongodbatlas/util"
+	"go.mongodb.org/atlas-sdk/v20231001002/admin"
+)
+
+var _ resource.ResourceWithConfigure = &ProjectRS{}
+var _ resource.ResourceWithImportState = &ProjectRS{}
+
+func NewSearchNodeRS() resource.Resource {
+	return &SearchNodeRS{
+		RSCommon: RSCommon{
+			resourceName: "search_node",
+		},
+	}
+}
+
+type SearchNodeRS struct {
+	RSCommon
+}
+
+type tfSearchNodeRSModel struct {
+	ID          types.String   `tfsdk:"id"`
+	ClusterName types.String   `tfsdk:"cluster_name"`
+	ProjectID   types.String   `tfsdk:"project_id"`
+	Specs       types.List     `tfsdk:"specs"`
+	StateName   types.String   `tfsdk:"state_name"`
+	Timeouts    timeouts.Value `tfsdk:"timeouts"`
+}
+
+type tfSearchNodeSpecModel struct {
+	InstanceSize types.String `tfsdk:"instance_size"`
+	NodeCount    types.Int64  `tfsdk:"node_count"`
+}
+
+var SpecObjectType = types.ObjectType{AttrTypes: map[string]attr.Type{
+	"instance_size": types.StringType,
+	"node_count":    types.Int64Type,
+}}
+
+func (r *SearchNodeRS) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Computed: true,
+			},
+			"cluster_name": schema.StringAttribute{
+				Required: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"project_id": schema.StringAttribute{
+				Required: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"specs": schema.ListNestedAttribute{
+				Validators: []validator.List{
+					listvalidator.SizeAtMost(1),
+					listvalidator.SizeAtLeast(1),
+				},
+				NestedObject: schema.NestedAttributeObject{
+					Attributes: map[string]schema.Attribute{
+						"instance_size": schema.StringAttribute{
+							Required: true,
+						},
+						"node_count": schema.Int64Attribute{
+							Required: true,
+						},
+					},
+				},
+				Required: true,
+			},
+			"state_name": schema.StringAttribute{
+				Computed: true,
+			},
+			"timeouts": timeouts.Attributes(ctx, timeouts.Opts{
+				Create: true,
+				Update: true,
+				Delete: true,
+			}),
+		},
+	}
+}
+
+const defaultSearchNodeTimeout time.Duration = 3 * time.Hour
```

Comment:
[q] is this valid for all the operations? Why is 3 hours

Reply:
looked into similar resources to define this default value in the create/update/delete operations:

- [advanced cluster](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/mongodbatlas/resource_mongodbatlas_advanced_cluster.go#L317-L319)
- [advanced cluster](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/mongodbatlas/resource_mongodbatlas_cluster.go#L351-L353)
- [search index](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/mongodbatlas/resource_mongodbatlas_search_index.go#L31-L33)